tfc-dump.pl

```diff
@@ -30,140 +69,163 @@
+    $curl_cmd   = "curl $curl_headers $curl_query";
+    $jq_cmd     = "jq '.data.id'";
+
+    $tfc_workspace_id = slurp_string_output("$curl_cmd | $jq_cmd");
+    $tfc_workspace_id =~ s/"//g;
+    
+    $workspace_list{$tfc_workspace_name} = $tfc_workspace_id;
 }
 else {	# All workspaces desired
-	my $tfc_ops_cmd = "tfc-ops workspaces list --organization ${tfc_org_name} --attributes name,id";
-
-	my @result = `$tfc_ops_cmd`;
-
-	# tfc-ops prints two header lines before the data we want to see.
-	shift(@result);		# remove "Getting list of workspaces ..."
-	shift(@result);		# remove "name, id"
-	chomp(@result);		# remove newlines
-
-	my $name;
-	my $id;
-	foreach (@result) {
-		($name, $id) = split(/, /, $_);
-		$workspace_list{$name} = $id;
-	}
+    my $tfc_ops_cmd = "tfc-ops workspaces list --organization ${tfc_org_name} --attributes name,id";
+    my @result = slurp_array_output($tfc_ops_cmd);
+    if ($?) {
+        my $error = "Failed to list workspaces";
+        error_to_sentry($error);
+        die "$error\n";
+    }
+
+    # tfc-ops prints two header lines before the data we want to see.
+    shift(@result);		# remove "Getting list of workspaces ..."
+    shift(@result);		# remove "name, id"
+
+    foreach (@result) {
+        my ($name, $id) = split(/, /, $_);
+        $workspace_list{$name} = $id;
+    }
 }
 
 # Dump the workspace and variable data to files.
-
 foreach (sort keys %workspace_list) {
-
-	# Dump the workspace info
-	$curl_query = "\"https://app.terraform.io/api/v2/workspaces/$workspace_list{$_}\"";
-	$curl_cmd   = "curl $curl_headers --output $_-attributes.json $curl_query";
-	system($curl_cmd);
-
-	# Dump the variables info
-	$curl_query = "\"https://app.terraform.io/api/v2/workspaces/$workspace_list{$_}/vars\"";
-	$curl_cmd   = "curl $curl_headers --output $_-variables.json $curl_query";
-	system($curl_cmd);
+    # Dump the workspace info
+    $curl_query = "\"https://app.terraform.io/api/v2/workspaces/$workspace_list{$_}\"";
+    $curl_cmd   = "curl $curl_headers --output $_-attributes.json $curl_query";
+    if (system($curl_cmd) != 0) {
+        my $error = "Failed to dump workspace attributes for $_";
+        error_to_sentry($error);
+        print STDERR "$error\n";
+    }
+
+    # Dump the variables info
+    $curl_query = "\"https://app.terraform.io/api/v2/workspaces/$workspace_list{$_}/vars\"";
+    $curl_cmd   = "curl $curl_headers --output $_-variables.json $curl_query";
+    if (system($curl_cmd) != 0) {
+        my $error = "Failed to dump workspace variables for $_";
+        error_to_sentry($error);
+        print STDERR "$error\n";
+    }
 }
 
 # Dump the variable sets data to files.
-
-### WARNING ###
-#
-# This code assumes that all of the TFC Variable Sets are contained within
-# the first result page of 100 entries.  This was true for SIL in August 2024.
-#
-####
-
 my @vs_names;
 my @vs_ids;
-my $pg_size = 100;
+my $pg_size = 20;  # Default page size per Hashicorp documentation
```

Comment:
It is probably better to leave this at 100. Multiple URL calls means the script runs slower do to internet lag.